feat(french): Add French translation for the README file

Venatum opened the pull request:
A comprehensive French translation of the README file has been added, containing all sections, guides, and resources included in the original document. This improves accessibility for French-speaking users.

## Review the Contributing Guidelines

Before submitting a pull request, verify it meets all requirements in the [Contributing Guidelines](https://github.com/donnemartin/system-design-primer/blob/master/CONTRIBUTING.md).

### Translations

See the [Contributing Guidelines](https://github.com/donnemartin/system-design-primer/blob/master/CONTRIBUTING.md).  Verify you've:

Notify here: https://github.com/donnemartin/system-design-primer/issues/250

Venatum committed:
feat(french): Add French translation for the README file

A comprehensive French translation of the README file has been added, containing all sections, guides, and resources included in the original document. This improves accessibility for French-speaking users.

Venatum committed:
feat(solutions): add French README for Mint.com system design

Venatum committed:
feat(solutions): add French README for Pastebin.com (or Bit.ly) system design

Venatum committed:
feat(solutions): add French README for Query-Cache system design

Venatum committed:
feat(solutions): add French README for Amazon's sales rank system design

Venatum committed:
feat(solutions): add French README for AWS scale rank system design

Venatum committed:
feat(solutions): add French README for social network rank system design

Venatum committed:
feat(solutions): add French README for Twitter rank system design

Venatum committed:
feat(solutions): add French README for Web Crawler system design

Venatum committed:
style(README-fr): fix formatting and minor language inconsistencies

Refined line breaks, removed redundant spaces, and improved text structure for better readability.

Venatum committed:
style(mint, README-fr): align table schemas and fix grammar in RPC note

Venatum committed:
refactor(docs): improve formatting and consistency in French README files

- Adjusted table formatting for better readability.
- Updated iterative design process to use numbered lists.
- Fixed a typo in the Pastebin README ("contenue" -> "contenu").

Venatum committed:
style(web_crawler/twitter): improve table formatting and code block consistency in README-fr.md

spuyet commented:
Looks good but a bit painful to read to be honest as there is a lot of changes in one single PR.

Venatum commented:
> Looks good but a bit painful to read to be honest as there is a lot of changes in one single PR.

Yeah, I know. I've translated all the README files I could find, but the part with the links and table of contents still needs some reviewing and polishing

Venatum committed:
style(README-fr): fix anchor links and improve naming consistency in French README

- Updated multiple anchor links for accuracy and compatibility.
- Improved section naming and formatting to enhance readability and consistency.

Venatum commented:
> > Looks good but a bit painful to read to be honest as there is a lot of changes in one single PR.
> 
> Yeah, I know. I've translated all the README files I could find, but the part with the links and table of contents still needs some reviewing and polishing

Hello 👋 
Sorry for the delay, the links should be fixed
